File: 1195-distribute-candies-to-people/distribute-candies-to-people.ts
function distributeCandies(candies: number, num_people: number): number[] {
  let remainCandies = candies;
  let distributionCount = 1;
  let currentIndex = 0;
  const result: number[] = new Array(num_people).fill(0);

  while (remainCandies > 0) {
    if (remainCandies <= distributionCount) {
      result[currentIndex] += remainCandies;
      break;
    } else {
      result[currentIndex] += distributionCount;
      remainCandies -= distributionCount;
    }

    currentIndex = currentIndex >= num_people - 1 ? 0 : currentIndex + 1;
    distributionCount++;
  }

  return result;
}

File: 1195-distribute-candies-to-people/distribute-candies-to-people.go
func distributeCandies(candies int, num_people int) []int {
	remainingCandies := candies
	ditributionCount := 1
	currentIndex := 0
	result := make([]int, num_people)

	for remainingCandies > 0 {
		if remainingCandies <= ditributionCount {
			result[currentIndex] += remainingCandies
			return result
		} else {
			result[currentIndex] += ditributionCount
			remainingCandies -= ditributionCount
		}
		currentIndex = (currentIndex + 1) % num_people
		ditributionCount++
	}
	return result
}